### ISSUE TITLE
[BUG] [ADDON] [Karpenter] Error with instance profile

### ISSUE THREAD

**Issue by allamand** [aws-quickstart/cdk-eks-blueprints#317]
Trying to deploy Karpenter with AddOn, and let Karpenter create the default provisioner:

```
    const provisionerSpecs = {
      'node.kubernetes.io/instance-type': ['m5.2xlarge'],
      'topology.kubernetes.io/zone': ['eu-west-1a'],
      'kubernetes.io/arch': ['amd64', 'arm64'],
      'karpenter.sh/capacity-type': ['spot', 'on-demand'],
    };
```

>Note, not all fields should be mandatory as karpenter as sane defaults. Also specifying zone is problematic for multi-region setup with the blueprint.

I see this error in Karpenter:
```
manager 2022-03-22T17:51:39.209Z    ERROR    controller.provisioning    Could not launch node, launching instances, with fleet error(s), InvalidParameterValue: Value (ssp-pipeline-stack/ssp-dev/ssp-dev-blueprint/ssp-dev-blueprint/karpenter-instance-profile [AWS::IAM::InstanceProfile]) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name    {"commit": "5047f3c", "provisioner": "default"}
```

The generated provisioner retrive in the cluster:
```
spec:
  kubeletConfiguration: {}
  limits: {}
  provider:
    apiVersion: extensions.karpenter.sh/v1alpha1
    instanceProfile: ssp-pipeline-stack/ssp-dev/ssp-dev-blueprint/ssp-dev-blueprint/karpenter-instance-profile
      [AWS::IAM::InstanceProfile]
    kind: AWS
    securityGroupSelector:
      kubernetes.io/cluster/ssp-dev-blueprint: '*'
    subnetSelector:
      kubernetes.io/cluster/ssp-dev-blueprint: '*'
  requirements:
  - key: node.kubernetes.io/instance-type
    operator: In
    values:
    - m5.2xlarge
  - key: kubernetes.io/arch
    operator: In
    values:
    - amd64
    - arm64
  - key: karpenter.sh/capacity-type
    operator: In
    values:
    - spot
    - on-demand
```

I try to remove the `[AWS::IAM::InstanceProfile]` part  but then I got this error

```
│ manager 2022-03-22T18:15:45.230Z    ERROR    controller.provisioning    Could not launch node, launching instances, with fleet error(s), InvalidParameterValue: Value (ssp-pipeline-stack/ssp-dev/ssp-dev-bluepr │
│ int/ssp-dev-blueprint/karpenter-instance-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name
```

> also version is 0.5.3, we should bump to 0.7.2

**Comment by youngjeong46** [comment]
@allamand, I will try reproducing the error and get back to you on this issue.

**Comment by suhussai** [comment]
I ran into this issue as well. I think this is the pertinent line in the code:

https://github.com/aws-quickstart/cdk-eks-blueprints/blob/f51c11337607ccd3c0bc98295ab2ce84a77d322d/lib/addons/karpenter/index.ts#L109

I think that line should be:
```
instanceProfile: karpenterInstanceProfile.ref
```
As an intermediate solution, you could manually edit the provisioner created by the addon and edit the `instanceProfile` field to something like `KarpenterNodeInstanceProfile-${CLUSTER_NAME}`

I think the long-term fix for this should be to allow users to pass in an instance profile they'd like to use since the default one created by the addOn may not fit their needs.

**Comment by youngjeong46** [comment]
This issue is in progress - will be a PR to address this and #325.

**Comment by youngjeong46** [comment]
@allamand the PR is merged, expect this to show up as part of v1.0 release. Closing the issue.
